primefaces-integration-tests/src/test/java/org/primefaces/integrationtests/fileupload/FileUpload011Test.java

```diff
@@ -0,0 +1,251 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2009-2021 PrimeTek
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.primefaces.integrationtests.fileupload;
+
+import java.io.File;
+import org.json.JSONObject;
+import org.junit.jupiter.api.*;
+import org.openqa.selenium.support.FindBy;
+import org.primefaces.selenium.AbstractPrimePage;
+import org.primefaces.selenium.PrimeSelenium;
+import org.primefaces.selenium.component.DataTable;
+import org.primefaces.selenium.component.FileUpload;
+
+/**
+ * Tests advanced single file upload.
+ * p:fileUpload mode=advanced auto=false multiple=false dragDropSupport=false maxChunkSize="1024k"
+ */
+@Tag("SafariExclude") // Selenium SafariDriver does not support file uploads
+public class FileUpload011Test extends AbstractFileUploadTest {
+
+    @Test
+    @Order(1)
+    public void testAdvancedSingleChunkedUpload(Page page) {
+        // Arrange
+        FileUpload fileUpload = page.fileupload;
+        Assertions.assertEquals("", fileUpload.getValue());
+
+        // Act
+        File file = locateClientSideBigFile("primefaces-integration-tests-sources.jar");
```

Comment:
This is a nitpick but shouldn't locateCLientSideBiFile be in the `Arrange` section since you are setting up data?

Reply:
I think `locateClientSideBigFile` belongs to `Act`. It mimics the user action to select a file in the OS-dependent file dialog. Maybe it should be renamed to `chooseClientSideBigFile`. 
Btw. Please have a closer look into `AbstractFileUploadTest.locateClientSideBigFile`. It effectively picks files from the `primefaces-integration-tests/target` folder. I'm not sure if tha is a good idea. But I did not want to add random big files to the resources folder, just for test purpose.

Reply:
No problem!